### PR TITLE
Features/linking-play-challenge

### PR DESCRIPTION
**Pull request by lkihuynh**
- linked play-challenge frontend and backend; 
- shuffled definitions; 
- renamed view-challenge to play-challenge; 
- removed dummy data; 
- implemented .get/published;

### CODE REVIEW

File: src/backend/src/routes/Challenges.ts [hunk 1]
```diff
@@ -49,24 +51,47 @@ router.post("/draft", async (req: Request, res: Response) => {
...
+            a[i] = a[j];
+            a[j] = x;
+        }
+        return a;
+    }     
+
     try {
         // TODO: The frontend should receive possible definitions in a random order,
         // but also have knowledge on which definition is correct.
         // Check the "UnknownChallenge" type in backend/src/shared/types
```
Comment (WilliamChan4) [inline src/backend/src/routes/Challenges.ts:72]: You should have a function in the PublishedChallenge class which gets all the definitions, then returns them as a shuffled array. The unknownChallenge class that Will Coulter had before should just be an interface or a normal typescript class which you then make and send objects of to the frontend. 

The entire definition the user selected should then be sent back to the backend to be checked against the real definition stored in the database. 

This way, the entire thing is stateless and the real answer is never leaked to users.

File: src/backend/src/routes/Challenges.ts [hunk 2]
```diff
@@ -78,10 +103,45 @@ router.get("/published", async (req: Request, res: Response) => {
...
  */
+
 router.post("/published", async (req: Request, res: Response) => {
-    try {
+    try {      
+        const challenge_id = req.body.challenge_id;
+        console.log("challenge id", challenge_id);
+        const answer = req.body.answer;
+        const maybeChallenge = await PublishedChallengeModel.findById(challenge_id).exec();
+        const user:User = req.user! as User;
```
Comment (WilliamChan4) [inline src/backend/src/routes/Challenges.ts:119]: Req.user is of type DocumentType<User>. You need to import DocumentType from typegoose and User from the User model.

File: src/backend/src/routes/Challenges.ts (hunk 1 above)
Comment (lkihuynh) [reply src/backend/src/routes/Challenges.ts:72]: Okay! Thanks for the tip

**Comment by WilliamChan4** [comment]
Due to @william-coulter's pull request, this pull request would be overwritten if that merged into features/linking-frontend-to-backend. Therefore, I'm merging this with that branch so that @lkihuynh 's contribution to the project is not lost.

**Pull request merged**